app/Http/Controllers/AcceptancesController.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Acceptance;
+use App\Submission;
+use Illuminate\Http\Request;
+
+class AcceptancesController extends Controller
+{
+    public function store(Request $request)
+    {
+        $submission = Submission::findOrFail($request->input('submissionId'));
+        if (auth()->user()->id != $submission->talkRevision->talk->author_id) {
+            return response('', 401);
+        }
+
+        $acceptance = Acceptance::createFromSubmission($submission);
+
+        return response()->json([
+            'status' => 'success',
+            'message' => 'Talk Accepted!',
+            'acceptanceId' => $acceptance->id,
+        ]);
+    }
+
+    public function destroy($id)
+    {
+        $acceptance = Acceptance::findOrFail($id);
+        if (auth()->user()->id != $acceptance->submission->talkRevision->talk->author_id) {
+            return response('', 401);
+        }
+
+        $acceptance->submission->removeAcceptance();
```

Comment:
What if this were a hook on `submission->delete()`? Any time I delete a submission, null out the acceptance_id on its related submission?

Reply:
I'm a little confused. I think what you're mean is add a hook so that any time an acceptance is deleted it removes the `acceptance_id` on it's corresponding submission?

Reply:
Whoops! Yes. I meant, could you hook on delete acceptance, to null out the acceptance_id on its connected acceptance.